### PR TITLE
[Table sanity check] Integration test for the `users` table

### PR DESCRIPTION
**Pull request by akindyakov**
Fixes: #4915

### CODE REVIEW

File: osquery/tests/integration/tables/users.cpp [hunk 1]
```diff
@@ -14,34 +13,30 @@
...
-  //      {"directory", NormalType}
-  //      {"shell", NormalType}
-  //      {"uuid", NormalType}
-  //      {"type", NormalType}
-  //}
-  // 4. Perform validation
-  // EXPECT_TRUE(validate_rows(data, row_map));
+TEST_F(UsersTest, sanity) {
+  QueryData rows = execute_query("select * from users");
+  ASSERT_GT(rows.size(), 0ul); // There is must be at least one user
```
Comment (fmanco) [inline osquery/tests/integration/tables/users.cpp]: nit: extra `is`

File: osquery/tests/integration/tables/users.cpp [hunk 2]
```diff
@@ -14,34 +13,30 @@
...
+  auto row_map = ValidatatioMap{
+      {"uid", NonNegativeInt},
+      {"gid", NonNegativeInt},
+      {"uid_signed", IntType},
```
Comment (fmanco) [inline osquery/tests/integration/tables/users.cpp]: Doesn't these two  need to be non-negative too?

File: osquery/tests/integration/tables/users.cpp [hunk 3]
```diff
@@ -14,34 +13,30 @@
 
 #include <osquery/tests/integration/tables/helper.h>
 
+#include <osquery/logger.h>
```
Comment (fmanco) [inline osquery/tests/integration/tables/users.cpp]: Why do you need the logger?

File: osquery/tests/integration/tables/users.cpp [hunk 4]
```diff
@@ -14,34 +13,30 @@
...
+  auto row_map = ValidatatioMap{
+      {"uid", NonNegativeInt},
+      {"gid", NonNegativeInt},
+      {"uid_signed", IntType},
+      {"gid_signed", IntType},
+      {"username", NonEmptyString},
+      {"description", NormalType},
+      {"directory", NonEmptyString}, // The directory path could be fake
+      {"shell", FileOnDisk},
+      {"uuid", NormalType},
```
Comment (fmanco) [inline osquery/tests/integration/tables/users.cpp]: There is a `validUUID` check.

Comment (akindyakov) [reply osquery/tests/integration/tables/users.cpp]: Thanks!

File: osquery/tests/integration/tables/users.cpp (hunk 3 above)
Comment (akindyakov) [reply osquery/tests/integration/tables/users.cpp]: don't need it, you are right. Thanks!

File: osquery/tests/integration/tables/users.cpp (hunk 2 above)
Comment (akindyakov) [reply osquery/tests/integration/tables/users.cpp]: In the implementation of the tables there is only one difference between `uid` and `uid_signed` is sign. e.g. in linux implementation:
```c++
r["uid"] = BIGINT(pwd->pw_uid);
r["uid_signed"] = BIGINT((int32_t)pwd->pw_uid)
```
Something very similar for macos and freebsd.
But for windows:
```C++
   r["uid"] = INTEGER(getUidFromSid(sid));
   r["uid_signed"] = r["uid"];
```
May  be it is a bug. Do we need signed versions?

**Comment by akindyakov** [comment]
Done in other PR #5357

**Pull request closed without merging**